tests/main_tests.nim

```diff
@@ -48,8 +48,8 @@ suite "counter":
     check myCounter.value == 8
     myCounter.inc(0.5)
     check myCounter.value == 8.5
-    expect ValueError:
-      myCounter.inc(-1)
+    myCounter.inc(-1) # you shouldn't be doing this - but we don't want metrics to crash the app
```

Comment:
I totally do. Wrong API usage should crash the bloody application.

Reply:
not really - not for metrics and not for logging because these are meant to be used "risk-free" from anywhere in the application - if they need error handling, it should be done elsewhere, not in the call to increase the counter

Reply:
I disagree. This is not a toy library where wrong API usage can be ignored.

Reply:
If `myCounter.inc(-1)` is considered an invalid usage of the library, then it should raise a `Defect` and not a `ValueError`. The simplest treatment is to `doAssert` for it.